File: services/go/newApi/v1/users/getUser.go
package users

import (
	"errors"
	"newapi/utils"
	. "newapi/v1/commons"

	"github.com/gofiber/fiber/v2"
	"gorm.io/gorm"
)

func FindUser(c *fiber.Ctx) error {
	var username = c.Params("username")

	user, err := utils.GetUserPartial(username)

	if err != nil {
		if errors.Is(err, gorm.ErrRecordNotFound) {
			return c.Status(404).JSON(Response{
				Success: false,
				Message: "User not found.",
			})
		}

		return c.Status(500).JSON(Response{
			Success: false,
			Message: err.Error(),
		})
	}

	return c.JSON(Response{
		Success: true,
		Data:    user,
	})
}

File: apps/api/src/handlers/users/getUser.ts
import { eq } from "drizzle-orm";
import { db } from "../../db";
import { users } from "../../db/schemas";
import { FastifyImp, User } from "../../types";
import { USER_PUBLIC_OBJECT } from "../../utils/publicObjects";

export const getUser: FastifyImp<
  {
    Params: {
      username: string;
    };
  },
  unknown,
  true
> = async (request, reply) => {
  const { username } = request.params;

  const user: User =
    (
      await db
        .select(USER_PUBLIC_OBJECT)
        .from(users)
        .where(eq(users.username, username))
    )[0] ?? null;

  if (!user) {
    reply.status(404).send({
      success: false,
      error: {
        code: "not_found",
        message: "User not found",
      },
    });
  }

  reply.send({
    success: true,
    data: user,
  });
};
